libraries/dialogs/src/main/java/com/trendyol/uicomponents/dialogs/list/DialogListAdapter.kt

```diff
@@ -0,0 +1,67 @@
+package com.trendyol.uicomponents.dialogs.list
+
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import com.trendyol.dialog.R
+import com.trendyol.dialog.databinding.ItemListBinding
+import com.trendyol.uicomponents.dialogs.inflate
+
+internal class DialogListAdapter(
+    private val showItemsAsHtml: Boolean
+) : RecyclerView.Adapter<DialogListAdapter.ItemViewHolder>() {
+
+    private var items: List<Pair<Boolean, CharSequence>> = emptyList()
+    var onItemSelectedListener: ((Int) -> Unit)? = null
+
+    fun setItems(list: List<Pair<Boolean, CharSequence>>) {
+        items = items.toMutableList().apply {
```

Comment:
using toMutableList() duplicates a current list. 
Please prefer **MutableList** as a type of any list if you need to modify it.